pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -255,57 +276,64 @@ func (t *Tracer[Event]) EventCallback(event any) {
 }
 
 func (t *Tracer[Event]) AttachContainer(container *containercollection.Container) error {
+	// FIXME: AttachContainer called before t.Tracer is initialized
+	if t == nil {
+		return nil
+	}
 	return t.Attach(container.Pid)
 }
 
 func (t *Tracer[Event]) DetachContainer(container *containercollection.Container) error {
+	// FIXME
+	if t == nil {
+		return nil
+	}
 	return t.Detach(container.Pid)
 }
 
 func (t *Tracer[Event]) GetMap(name string) *ebpf.Map {
 	return t.collection.Maps[name]
 }
 
-func (t *Tracer[Event]) listen(
-	rd *perf.Reader,
-	baseEvent func(ev types.Event) *Event,
-	processEvent func(rawSample []byte, netns uint64) (*Event, error),
-	eventCallback func(*Event),
-) {
+func (t *Tracer[Event]) Listen() {
+	go t.listen()
+}
```

Comment:
Any reason not to combine this with `Run()`. I image builtin gadgets could have something like:

```golang
	err = networkTracer.Run(spec, types.Base, parseDNSEvent)
	if err != nil {
		return fmt.Errorf("setting network tracer spec: %w", err)
	}
```